src/tsconfig.json

```diff
@@ -4,7 +4,14 @@
 		"removeComments": false,
 		"preserveConstEnums": true,
 		"sourceMap": false,
-		"outDir": "../out"
+		"outDir": "../out",
+		"target": "es6",
+		"lib": [
+			"dom",
+			"es5",
+			"es2015.iterable",
```

Comment:
@DanielRosenwasser @mjbvz I cannot get this to work with `es2015.iterable`. Without it all destructing and for-of is illegal, independent of the `downlevelIteration`-switch. The downside is that `es2015.iterable` drags in all of `Symbol` which we don't want (in the monaco editor and below)

Reply:
Quick conservative workaround:

```ts
declare var Symbol: {
  iterator: symbol;
};
```

Reply:
Is the Monaco editor also being compiled to ES6 or does it have its own tsconfig.json file?

Reply:
When you use `--target es6` we use native destructuring and iteration, which requires `Symbol.iterator` at runtime, therefore we type-check that the array being destructured/iterated has a `[Symbol.iterator]()` method that can be used.

If you wan to leave this out of the tsconfig.json file, one option is to add a `/// <reference lib="es2015.iterable" />` to the files that need iteration.

Reply:
@rbuckton we do have a separate `tsconfig.monaco.json` which targets ES5.